Zeek Runner User Preference

jameskerr opened the pull request:
The ZQD class will now look in the global preferences for a user specified zeek runner. This means that if a user changes the value, it will not be in effect until the app is restarted. Because of this, whenever the value changes, we show a message "Restart required."

<img width="972" alt="image" src="https://user-images.githubusercontent.com/3460638/81755597-0d611100-946e-11ea-8f54-336f65b8c413.png">

<img width="314" alt="Screen Shot 2020-05-12 at 4 27 25 PM" src="https://user-images.githubusercontent.com/3460638/81755612-17830f80-946e-11ea-872b-36b4d7e32931.png">

fixes: #741

jameskerr commented on src/css/_settings-modal.scss:
```diff
@@ -13,14 +13,14 @@
     }
 
     select {
-      border: none;
```
@mason-fish border gone!

jameskerr commented on src/js/components/Preferences/Preferences.js:
```diff
@@ -46,6 +47,7 @@ export default function Preferences() {
         <form ref={setForm} className="settings-form">
           <Timezone config={prefsForm.timeZone} />
           <TimeFormat config={prefsForm.timeFormat} />
+          <ZeekRunner config={prefsForm.zeekRunner} />
```
Almost a direct copy and paste from JSONTypeConfig. Potential for abstracting soon.

jameskerr replied:
Update: Did abstract this to a FileInput component used for the preferences modal.

jameskerr commented on src/js/zqd/zqd.js:
```diff
@@ -78,33 +79,29 @@ function zqdCommand(): string {
-  }
-
-  if (!pathExistsSync(zeekRunner)) {
-    throw new Error("zeek runner not present at " + zeekRunner)
```
I removed this thow since the value is now user specified. If they provided invalid input, the app would not start.

Instead, for now, if there is a bad zeekrunner value set, the app starts up fine, but we display an error when they try to import a pcap.

Here is what the error will look like if the zeek runner does not exist:

<img width="509" alt="Screen Shot 2020-05-12 at 4 11 34 PM" src="https://user-images.githubusercontent.com/3460638/81755884-e951ff80-946e-11ea-846a-0d5e8e81ad9c.png">

And here is if the zeek runner file is present but invalid

<img width="503" alt="Screen Shot 2020-05-12 at 4 14 25 PM" src="https://user-images.githubusercontent.com/3460638/81755897-f373fe00-946e-11ea-9a60-3f7e970866e8.png">

alfred-landrum replied:
Thanks for the screenshots - I can improve those error messages, filed:
https://github.com/brimsec/zq/issues/767

jameskerr commented:
# Update

Added doc links to the zeek runner and json type config wiki pages.

<img width="728" alt="image" src="https://user-images.githubusercontent.com/3460638/81856604-e827dd80-9515-11ea-8990-0d4b35214f50.png">

Added a doc link and an explanation in the ingest errors modal.

<img width="702" alt="Screen Shot 2020-05-13 at 12 26 45 PM" src="https://user-images.githubusercontent.com/3460638/81856679-0097f800-9516-11ea-9ce3-28d0a071df35.png">

mason-fish commented on src/js/components/IngestWarningsModal.js:
```diff
@@ -31,7 +33,15 @@ export default function IngestWarningsModal() {
       <TextContent>
         {warnings.length ? (
-          <pre className="output">{warnings.join("\n")}</pre>
+          <>
```
ah cool didn't know about fragments

mason-fish commented on src/js/components/common/Link.js:
```diff
@@ -0,0 +1,18 @@
+
+type Props = {|href: string, children: *|}
+
+export default function Link({href, children}: Props) {
```
Nice, a new common component! I had also included these in the common/index.js so that we can import/export the same way we do with the `lib` package.

mason-fish commented on src/js/zqd/zqd.js:
```diff
@@ -78,33 +79,29 @@ function zqdCommand(): string {
-  if (!pathExistsSync(zeekRunner)) {
-    throw new Error("zeek runner not present at " + zeekRunner)
-  }
+  const precedence = [
```
`s/precedence/preference` ?

jameskerr replied:
I meant precedence here. This is the lookup order for the zeek runner path. First check the ENV, then the userPref, then the default.
